File: frontend/src/types/stock.ts
export interface Stock {
    ticker: string;
    name: string;
    current_price: number;
    currency: string;
    market_cap: number;
    day_high: number;
    day_low: number;
}

export interface HistoricalDataPoint {
    date: string;
    close: number;
}

File: backend/app/models/stock.py
# app/models/stock.py

from pydantic import BaseModel

class StockResponse(BaseModel):
    ticker: str
    name: str | None = None
    current_price: float | None = None
    currency: str | None = None
    market_cap: int | None = None
    day_high: float | None = None
    day_low: float | None = None

class HistoricalDataPoint(BaseModel):
    date:str
    close:float